Instancer prototypes

andrewkaufman opened the pull request:
This makes Gaffer's `Instancer` a bit easier to work with, by organizing the _NodeEditor_, setting some userDefaults, and renaming the `instances` plug to `prototypes` (matching USD naming).

I've also added a scale V3f to the `OSLObject` "Standard" menu to make it easier to setup primvars matching the new `Instancer` defaults.

There could be an argument to use `orient` rather than `orientation` (matching Houdini defaults), in which case we'd also want to update our `Orientation` node. Similarly, `id` is a common standard (as opposed to `instanceId`). If you feel either of those are worthwhile, just let me know and I'll push a change.

johnhaddon commented:
Thanks Andrew - looks good.

Now we've renamed `instances->prototypes` does it also make sense to rename `index->prototypeIndex`? I have to think a bit to realise the distinction between `index` and `id`, and solidifying the connection between index and prototypes could help a fair bit. It'd also be good to trickle down the renaming further through the implementation in Instancer.cpp - in particular `EngineData::instanceIndex()` is even more misleading now, and could be renamed to `prototypeIndex()`.

Since the plug renaming breaks ABI, it'll need to be targeted to master rather than 0.55.

Cheers...
John

andrewkaufman commented:
Ok if I squash those changes into the appropriate commits or would you prefer separate new commits?

johnhaddon commented:
Squash away!

andrewkaufman commented:
@johnhaddon I need 0.55_maintenance merged forward into master before I can change the base branch.

andrewkaufman commented:
> rename index->prototypeIndex and trickle down the renaming further through the implementation in Instancer.cpp

Addressed in 6d9addb04c4e54470494c775caba9520ac494632. Would be good to get an eye on it, there were a lot of uses of the word 'instance' in that file, and I think I changed all the right ones, but its hard to be sure... most of the changes are pretty trivial, but take a closer look at the compute of `( output == instanceChildNamesPlug() )` and `computeBranchSet` in particular.

johnhaddon commented:
> Would be good to get an eye on it, there were a lot of uses of the word 'instance' in that file, and I think I changed all the right ones, but its hard to be sure.

I think this omitted a few, but it was easier to merge this and put up a followup PR than try and explain. Hopefully this makes sense :

https://github.com/GafferHQ/gaffer/pull/3458
